File: Sample Codes/Traversal.java
class Node{
	Object item;
	Node left;
	Node right;
	Node(Object item){
		this.item = item;
		left = right = null;
	}

	static void preorder(Node node){
		if(node == null) return;
		System.out.println(node.item);
		preorder(node.left);
		preorder(node.right);
	}

	static void inorder(Node node){
		if(node == null) return;
		inorder(node.left);
		System.out.println(node.item);
		inorder(node.right);
	}

	static void postorder(Node node){
		if(node == null) return;
		postorder(node.left);
		postorder(node.right);
		System.out.println(node.item);
	}
}

class Test{
	public static void main(String[] args) {
		Node a = new Node("a");
		Node b = new Node("b");
		Node c = new Node("c");
		Node d = new Node("d");

		a.left = b;
		a.right = c;
		b.right = d;

		Node.preorder(a);
		Node.inorder(a);
		Node.postorder(a);
	}
}

File: Sample Codes/Traversal.py
class Node:
	item = 0
	left = None
	right = None
	
def preorder(node):
	if not node:
		return

	print node.item
	preorder(node.left)
	preorder(node.right)

def inorder(node):
	if not node:
		return

	inorder(node.left)
	print node.item
	inorder(node.right)

def postorder(node):
	if not node:
		return

	postorder(node.left)
	postorder(node.right)
	print node.item

a = Node()
b = Node()
c = Node()
d = Node()

a.item = "A"
b.item = "B"
c.item = "C"
d.item = "D"

a.left = b
a.right = c
b.right = d

print "Preorder:"
preorder(a)
print "Inorder:"
inorder(a)
print "Postorder:"
postorder(a)

